tests/integration_test.py

```diff
@@ -296,3 +297,26 @@ def test_that_invalid_sized_buffers_are_dropped(tmpdir):
         buffers = [x.split() for x in output.strip().split('\n')]
 
         assert 35 < len(buffers) < 65
+
+
+def parse_gst_timestamp(timestamp):
+    m = re.match(r"(\d+):(\d\d):(\d\d\.\d+)", timestamp)
+    if m:
+        return (int(m.group(1)) * 60 * 60 + int(m.group(2)) * 60
+                + float(m.group(3)))
+
+
+def test_that_timestamps_are_preserved(tmpdir):
+    with pulsevideo_ctx(tmpdir, source_pipeline='videotestsrc is-live=false'):
+        try:
+            output = subprocess.check_output(
+                ['gst-launch-1.0', '-q', 'pulsevideosrc',
+                 'bus-name=com.stbtester.VideoSource.test', '!', 'identity',
+                 'error-after=11', '!', 'checksumsink'])
+        except subprocess.CalledProcessError as e:
+            output = e.output
+    buffers = [x.split() for x in output.strip().split('\n')]
+    timestamps = [parse_gst_timestamp(x[0]) for x in buffers]
+    assert len(timestamps) == 10
+    for n in range(1, 9):
+        assert abs(timestamps[n] + 0.1 - timestamps[n + 1]) < 1e-9
```

Comment:
This testcase would be clearer if the videotestsrc framerate (which defaults to 10fps) is stated explicitly in the source_pipeline.